exec/exec.go

```diff
@@ -110,52 +125,51 @@ func (cmd *cmdWrapper) SetStderr(out io.Writer) {
 
 // Run is part of the Cmd interface.
 func (cmd *cmdWrapper) Run() error {
-	return (*osexec.Cmd)(cmd).Run()
+	err := cmd.Cmd.Run()
+	return handleError(err)
 }
 
 // CombinedOutput is part of the Cmd interface.
 func (cmd *cmdWrapper) CombinedOutput() ([]byte, error) {
-	out, err := (*osexec.Cmd)(cmd).CombinedOutput()
-	if err != nil {
-		return out, handleError(err)
-	}
-	return out, nil
+	out, err := cmd.Cmd.CombinedOutput()
+	return out, handleError(err)
 }
 
 func (cmd *cmdWrapper) Output() ([]byte, error) {
-	out, err := (*osexec.Cmd)(cmd).Output()
-	if err != nil {
-		return out, handleError(err)
-	}
-	return out, nil
+	out, err := cmd.Cmd.Output()
+	return out, handleError(err)
 }
 
 // Stop is part of the Cmd interface.
 func (cmd *cmdWrapper) Stop() {
-	c := (*osexec.Cmd)(cmd)
-	if c.ProcessState.Exited() {
+	c := cmd.Cmd
+	if c.Process == nil {
 		return
 	}
+
 	c.Process.Signal(syscall.SIGTERM)
+
 	time.AfterFunc(10*time.Second, func() {
-		if c.ProcessState.Exited() {
-			return
+		if c.ProcessState != nil && !c.ProcessState.Exited() {
```

Comment:
How could this be nil if c.Process was not?  If possible, it deserves a comment

Reply:
just to be cautious. there is no documented connection between `c.Process` and `c.ProcessState`. But docs do say it is available after calling Wait or Run. It should be safe. I am okay to leave it as is.